stdlib/REPL/test/repl.jl

```diff
@@ -1696,6 +1696,41 @@ finally
     empty!(Base.Experimental._hint_handlers)
 end
 
+try # test the functionality of `UndefVarError_hint` against import clashes
+    @assert isempty(Base.Experimental._hint_handlers)
+    Base.Experimental.register_error_hint(REPL.UndefVarError_hint, UndefVarError)
+
+    fake_repl() do stdin_write, stdout_read, repl
```

Comment:
I don’t think you need a fake REPL for this, which will just mess up stacktraces if anything goes awry. Calling eval directly should be sufficient here and more reliable

Reply:
ah ok, I copied the previous test, which IIUC maybe also does not need a fake repl